src/bidiMapper/modules/log/LogManager.ts

```diff
@@ -104,23 +104,15 @@ export class LogManager {
         return;
       }
 
-      const argsPromise: Promise<Script.RemoteValue[]> =
-        realm === undefined
-          ? Promise.resolve(params.args as Script.RemoteValue[])
-          : // Properly serialize arguments if possible.
-            Promise.all(
-              params.args.map((arg) => {
-                return realm.serializeCdpObject(
-                  arg,
-                  Script.ResultOwnership.None
-                );
-              })
-            );
-      for (const browsingContext of realm.associatedBrowsingContexts) {
-        this.#eventManager.registerPromiseEvent(
-          argsPromise.then((args) => ({
-            kind: 'success',
-            value: {
+      try {
+        const args = await Promise.all(
+          params.args.map((arg) => {
+            return realm.serializeCdpObject(arg, Script.ResultOwnership.None);
+          })
+        );
+        for (const browsingContext of realm.associatedBrowsingContexts) {
+          this.#eventManager.registerEvent(
```

Comment:
I think this will change the order of events

Reply:
That's correct...